### PR TITLE
Raise ValueError when anything but map[string]string is used for Thrift

### PR DESCRIPTION
**Pull request by breerly**
cc @blampe @abhinav @junchaowu

### CODE REVIEW

File: tests/schemes/test_thrift.py [hunk 1]
```diff
@@ -1141,3 +1141,44 @@ def testString(request, response, proxy):
...
+@pytest.mark.gen_test
+@pytest.mark.callz
+def test_headers_should_be_a_map_of_strings():
+
+    tchannel = TChannel('client')
+
+    with pytest.raises(ValueError):
+        yield tchannel.thrift(
+            request=True,
+            headers={'key': 1},
```
Comment (blampe) [inline tests/schemes/test_thrift.py]:
```
@pytest.mark.parametrize("headers", [
    {'key': 1},
    ...
])
def test_headers_should_be_a_map_of_strings(headers):
    with ...:
        yield tchannel.thrift(
            ...
            headers=headers
        )
```

Comment (junchaowu) [inline tests/schemes/test_thrift.py]: +1

Comment (abhinav) [inline tests/schemes/test_thrift.py]: +1

**Comment by abhinav** [comment]
lgtm besides test parameterization

File: tests/schemes/test_thrift.py (hunk 1 above)
Comment (breerly) [inline tests/schemes/test_thrift.py]: yup

**Pull request merged**